x11-misc/gtkdialog/files/gtkdialog-0.8.3-fix-build-for-clang16.patch

```diff
@@ -0,0 +1,191 @@
+Clang16 will not allow implicit function declarations and implicit integers etc.
+This patch overhauls the source code for modern C.
+
+Bug: https://bugs.gentoo.org/875704
+Upstream PR: https://github.com/oshazard/gtkdialog/pull/81
+
+Signed-off-by: Pascal Jäger <pascal.jaeger@leimstift.de>
+
+--- a/src/actions.c
++++ b/src/actions.c
+@@ -36,6 +36,7 @@
+ #include "attributes.h"
+ #include "variables.h"
+ #include "tag_attributes.h"
++#include "gtkdialog_parser.h"
+ 
+ extern gchar *option_include_file;
+ 
+--- a/src/automaton.c
++++ b/src/automaton.c
+@@ -35,6 +35,7 @@
+ */
+ 
+ #include <gtk/gtk.h>
++#include <glib/gprintf.h>
+ 
+ #include "config.h"
+ #include "gtkdialog.h"
+@@ -87,6 +88,8 @@
+ #undef TOOLTIPS
+ 
+ extern gboolean option_no_warning;
++extern void push_widget(GtkWidget * widget, int widgettype);
++int instruction_execute(instruction command);
+ 
+ instruction *program = NULL;
+ int instruction_counter = 0;		/* The first available memory cell */
+@@ -323,8 +326,7 @@ void print_command(instruction command)
+     fflush(stdout);
+ }
+ 
+-void
+-print_program()
++void print_program()
+ {
+ 	gint pc;
+ 	instruction command;
+@@ -580,8 +582,7 @@ finalize:
+ }
+ 
+ 
+-int 
+-instruction_execute(instruction command)
++int instruction_execute(instruction command)
+ {
+ 	GList *element;
+ 	token Token;
+@@ -911,9 +912,11 @@ gboolean widget_moved(GtkWidget *widget,
+ 		//gtk_widget_set_usize(window, 
+ 		//		configure->width -20,
+ 		//		configure->height);
++#if HAVE_GTK==2
+ 		gtk_widget_set_uposition(GTK_WIDGET(window), 
+ 				configure->x,
+ 				configure->y);
++#endif
+ 	}
+ 	
+ 	fflush(stderr);
+--- a/src/gtkdialog.h
++++ b/src/gtkdialog.h
+@@ -65,5 +65,6 @@ gint get_program_from_variable(gchar *name);
+ void set_program_name(gchar *name);
+ gchar *get_program_name(void);
+ void load_styles_file(gchar *filename);
++int getnextchar(void);
+ 
+ #endif
+--- a/src/gtkdialog_parser.h
++++ b/src/gtkdialog_parser.h
+@@ -385,3 +385,8 @@ typedef union YYSTYPE
+ extern YYSTYPE gtkdialog_lval;
+ 
+ 
++extern int gtkdialog_error(char *c);
++extern void run_program(void);
++extern void print_program();
++extern int gtkdialog_lex (void);
++extern int gtkdialog_parse (void);
```

Comment:
I could not find that function anywhere in the codebase. I just made that prototype up. It somehow works. :question: :question: :question: :question:

Reply:
Yeah this is one of those yacc things and I think it's fine.